src/apps/lwaftr/lwaftr.lua

```diff
@@ -43,6 +44,43 @@ local function compute_binding_table_by_ipv4(binding_table)
    return ret
 end
 
+local function port_to_psid(port, reserved_ports_bit_count, psid_len)
+   local shift = 16 - (reserved_ports_bit_count + psid_len)
+   return band(rshift(port, shift), (2^psid_len)-1)
```

Comment:
Let's change to take shift as an argument.  Let's also add a case to "assert" that if reserved_ports_bit_count is greater than 0 then we assign a bogus PSID so that the lookup will fail.  Like this:

```lua
local function port_to_psid(port, psid_len, shift)
   local psid_mask = bit.lshift(1, psid_len)-1
   local psid = bit.band(bit.rshift(port, shift), psid_mask)
   if psid_len + shift < 16 then
      local reserved_ports_bit_count = 16 - psid_len - shift
      local first_allocated_port = bit.lshift(1,reserved_ports_bit_count)
      if port < first_allocated_port then psid = psid_mask + 1 end
   end
   return psid
end
```

Pretty sure the impact of the test will be small in the usual case that reserved_ports_bit_count is 0.